scr/main/java/bittercode/util/DBUtil.java

```diff
@@ -0,0 +1,39 @@
+package com.bittercode.util;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+import com.bittercode.constant.ResponseCode;
+import com.bittercode.model.StoreException;
+
+public class DBUtil {
+
+    private static Connection connection;
+
+    static {
+
+        try {
+
+            Class.forName(DatabaseConfig.DRIVER_NAME);
+            
+            connection = DriverManager.getConnection(DatabaseConfig.CONNECTION_STRING, DatabaseConfig.DB_USER_NAME,
+                    DatabaseConfig.DB_PASSWORD);
+        } catch (SQLException | ClassNotFoundException e) {
+
+            e.printStackTrace();
+
+        }
+
+    }// End of static block
+
+    public static Connection getConnection() throws StoreException {
+
+        if (connection == null) {
+            throw new StoreException(ResponseCode.DATABASE_CONNECTION_FAILURE);
+        }
+
+        return null;

```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Connection method returns null instead of connection</b></div>
<div id="fix">

The `getConnection()` method is returning `null` instead of the actual database connection. This could cause `NullPointerException` in code that uses this connection.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">

```
 @@ -35,7 +35,7 @@

         if (connection == null) {
             throw new StoreException(ResponseCode.DATABASE_CONNECTION_FAILURE);
         }
 -        return null;
 +        return connection;
```




```
 @@ -35,7 +35,7 @@

         if (connection == null) {
             throw new StoreException(ResponseCode.DATABASE_CONNECTION_FAILURE);
         }
 -        return null;
 +        return connection;
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/7#issuecomment-2527741727>#f9242c</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged